lib/features/warger_created/widgets/general_text_form_field.dart

```diff
@@ -0,0 +1,245 @@
+  final Alignment? align;
+  final Function(String)? validator;
+  final double? cursorHeight;
+  final TextAlign? textAlign;
+  final EdgeInsets? contentPadding;
+
+  const FormattedTextFields({
+    super.key,
+    this.title,
+    required this.keyInputType,
+    required this.textFieldController,
+    required this.textFieldHint,
+    this.textFieldLineSpan,
+    this.height,
+    this.width,
+    this.validator,
+    this.errorText,
+    this.textFieldMinLineSpan,
+    this.containerColor,
+    this.enableInteractiveSelection,
+    required this.noBorder,
+    this.autoFocus,
+    this.textInputAction,
+    required this.inputFormatters,
+    required this.onChangedFunction,
+    required this.errorTextActive,
+    required this.focusNode,
+    this.prefixIcon,
+    this.prefix,
+    this.suffixIcon,
+    this.textFont,
+    this.hintFont,
+    this.labelFont,
+    this.cursorHeight,
+    this.borderRadius,
+    this.cursorColor,
+    this.hintColor,
+    this.labelColor,
+    this.outLineBorderColor,
+    this.focusBorderColor,
+    this.textFontWeight,
+    this.hintFontWeight,
+    this.labelFontWeight,
+    this.obscureText,
+    this.labelBehaviour,
+    this.maxLength,
+    this.align,
+    this.onTap,
+    this.readOnly,
+    this.textAlign,
+    this.contentPadding,
+    this.onEditingComplete,
+  });
+
+  @override
+  State<FormattedTextFields> createState() => _FormattedTextFieldsState();
+}
+
+class _FormattedTextFieldsState extends State<FormattedTextFields> {
+  FocusNode _focusNode = FocusNode();
+  @override
+  void initState() {
+    super.initState();
+    _focusNode.addListener(() {
+      setState(() {}); // Update the UI when the focus changes
+    });
+  }
+
+  @override
+  void dispose() {
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          'inviteViaUsername'.tr(),
+          style: AppTheme.textMediumNormal.copyWith(
+              color: AppColors.blue950,
+              fontWeight: FontWeight.w500,
+              fontSize: 14),
+        ),
+        verticalSpace(12),
+        GestureDetector(
+          onTap: () => FocusManager.instance.primaryFocus!.requestFocus(),
+          child: Container(
+              height: widget.height ?? 72,
+              padding: EdgeInsets.only(left: 18),
+              alignment: widget.align ?? Alignment.centerLeft,
+              decoration: BoxDecoration(
+                color: AppColors.grayCool100,
+                borderRadius: BorderRadius.circular(12),
+              ),
+              child: _focusNode.hasFocus
+                  ? RichText(
+                      text: TextSpan(
+                        children: [
+                          TextSpan(
+                            text: 'wager.strk/',
+                            style: TextStyle(color: Colors.grey),
```

Comment:
use colors from `AppColor` and if it isn't there you can pick from ui and add there